tests/periph_eeprom/main.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2018 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Manual test application for the EEPROM peripheral drivers
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "shell.h"
+
+#include "periph/eeprom.h"
+
+#ifndef BUFFER_SIZE
+#define BUFFER_SIZE     (42U)
+#endif
+
+static char buffer[BUFFER_SIZE + 1];
+
+static int cmd_info(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+#ifdef EEPROM_START_ADDR
+    printf("EEPROM start addr:\t0x%08x\n", (int)EEPROM_START_ADDR);
+#endif
+    printf("EEPROM size:\t\t%i\n", (int)EEPROM_SIZE);
+
+    return 0;
+}
+
+static int cmd_read(int argc, char **argv)
+{
+    if (argc < 3) {
+        printf("usage: %s <pos> <count>\n", argv[0]);
+        return 1;
+    }
+
+    uint32_t pos = atoi(argv[1]);
+    uint8_t count = atoi(argv[2]);
+
+    if (!count) {
+        puts("Count should be greater than 0");
+        return 1;
+    }
+
+    if (count > BUFFER_SIZE) {
+        puts("Count exceeds buffer size");
+        return 1;
+    }
+
+    if (pos + count >= EEPROM_SIZE) {
+        puts("Failed: cannot read out of eeprom bounds");
+        return 1;
+    }
+
+    size_t ret = eeprom_read(pos, (uint8_t *)buffer, count);
+    buffer[count] = '\0';
```

Comment:
if you want to do this you must check for `count >= BUFFER_SIZE` above.

Reply:
nope, the buffer has an extra byte, see [here](https://github.com/RIOT-OS/RIOT/pull/8862/files#diff-0ec3102ff4d926d871687a7b42380deaR33)

Reply:
oops, oversaw that 😬